src/error.rs

```diff
@@ -32,6 +32,13 @@ pub(crate) enum Error {
   FileSearch { source: ignore::Error },
   #[snafu(display("Invalid glob: {}", source))]
   GlobParse { source: globset::Error },
+  #[snafu(display("Failed to parse hex string `{}`: {}", text, source))]
+  HexParse {
```

Comment:
I can see why you want `MagnetLinkParse` to have access to the link URL. But I think it's cleaner to put it inside every variant of `MagnetLinkParseError`---otherwise, we'd lose the `From` conversions.

Reply:
I'm also cool with putting it inside of the MagnetLinkParserError, I was thinking it be annoying.